NetKVM/Common/ParaNdis_TX.cpp

```diff
@@ -1100,3 +1100,61 @@ ULONG CNB::Copy(PVOID Dst, ULONG Length) const
 
     return Copied;
 }
+
+/*
+ * Copies data from an MDL chain to a destination buffer.
+ *
+ * @param Dst Pointer to the destination buffer where the data will be copied to.
+ * @param Length The maximum number of bytes to copy.
+ * @param Source Reference to a pointer to the current MDL in the chain. This MDL pointer
+ *               will be updated as the function traverses the MDL chain.
+ * @param Offset Reference to the current offset within the current MDL. This offset will
+ *               be updated as data is copied from the MDL.
+ * @return The total number of bytes copied to the destination buffer.
+ *
+ * Note: The function updates 'Source' to point to the next MDL in the chain after the
+ *       current MDL has been processed. If the end of the current MDL is reached before
+ *       copying 'Length' bytes, 'Source' will point to the next MDL for further copying.
+ *       'Offset' is updated to reflect the position within the current MDL after the copy
+ *       operation. If the end of the MDL is reached, 'Offset' will be set to zero to indicate
+ *       the start of the next MDL in the chain.
+*/
+ULONG CNB::CopyFromMdlChain(PVOID Dst, ULONG Length, PMDL &Source, ULONG &Offset) const
```

Comment:
nit: the procedure does not use CNB, so it is probably static member (if member)

Reply:
Okay, it has been modified.